src/main/java/org/betonquest/betonquest/quest/condition/variable/VariableCondition.java

```diff
@@ -0,0 +1,63 @@
+package org.betonquest.betonquest.quest.condition.variable;
+
+import org.betonquest.betonquest.api.logger.BetonQuestLogger;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.api.quest.condition.PlayerCondition;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.instruction.variable.VariableString;
+
+import java.util.regex.PatternSyntaxException;
+
+/**
+ * Checks if the variable value matches given pattern.
+ */
+public class VariableCondition implements PlayerCondition {
+
+    /**
+     * Custom {@link BetonQuestLogger} instance for this class.
+     */
+    private final BetonQuestLogger log;
+
+    /**
+     * The variable to compare with the regex.
+     */
+    private final VariableString variable;
+
+    /**
+     * The regex the variable must match.
+     */
+    private final VariableString regex;
+
+    /**
+     * The address of the variable.
+     */
+    private final String variableAddress;
```

Comment:
I think we could be a bit more specific here in the javadocs that variableAddress is only used for logging to specify where the issue actually is. May some further refactoring would make this information obsolet.